kartograf/merge.py

```diff
@@ -139,20 +164,37 @@ def general_merge(
     Merge lists of IP networks into a base file.
     """
     print("Merging extra prefixes that were not included in the base file.")
-    base = BaseNetworkIndex()
+    base_network_index = BaseNetworkIndex()
     with open(base_file, "r") as file:
         for line in file:
             pfx, _ = line.split(" ")
-            base.update(pfx)
+            base_network_index.update(pfx)
 
     df_extra = extra_file_to_df(extra_file)
 
-    extra_included = []
-    for row in df_extra.itertuples(index=False):
-        result = base.contains_row(row)
-        extra_included.append(result)
+    chunk_size = 50000
```

Comment:
The optimal chunk size will probably depend on the machine, can you try if you can get similar or better results with dynamic picking of the chunk size with something like this (untested):

```python
def pick_chunk_size(n_rows: int, workers: int | None = None,
                    tasks_per_worker: int = 16,
                    min_chunk: int = 10_000,
                    max_chunk: int = 200_000) -> int:
    if workers is None:
        workers = os.cpu_count() or 4
    num_tasks = max(1, workers * tasks_per_worker)
    chunk = math.ceil(n_rows / num_tasks)
    return max(min_chunk, min(max_chunk, chunk))
```